### DISCUSSION TITLE
Are compound unique constraints supported when nesting?

### DISCUSSION THREAD

**Discussion by dugramen** [prisma/prisma#22543] [Q&A]
### Question

I have videos which point to a sourceId (like a youtube id). And another table, scores, for scoring videos.
```
model Video {
  id       String   @id @default(cuid())
  sourceId String
  provider Provider @default(YOUTUBE)
  scores   Score[]

  @@unique([sourceId, provider])
}

model Score {
  id      String @id @default(cuid())
  videoId String
  userId  String
  score   Int
  video   Video @relation(fields: [videoId], references: [id])

  @@unique([videoId, userId])
}
```
But it appears I can't update the score with an upsert query, without knowing Video.id. Even if I know sourceId and provider, which have a compound unique rule, a query like this just gives typescript errors
```
await prisma.score.upsert({
  where: { 
    userId: session.user.id,
    video: {
      sourceId: ytId,
      provider: "YOUTUBE"
    },
  },
  ...
})
```
Below is the only thing allowed, but it requires knowing videoId
```
await prisma.score.upsert({
  where: { 
    videoId_userId: {
      userId: session.user.id,
      videoId: 'Some video id'
    }
  },
  ...
})
```
I know this can be done with multiple queries, but I want to be able to do it in a single upsert. Is there a different schema design that would let me do this, or maybe I can still do it with this design?

**Comment by ludralph** [comment]
Hi @dugramen 👋 

Thank you for raising this question. 

The upsert issue you are facing is a known limitation as can be seen in this open [issue](https://github.com/prisma/prisma/issues/8324).  There's no way to do this with a single query yet. you might need to perform a separate query to retrieve the `videoId` based on the `sourceId` and `provider` before you can execute the upsert operation. If you want to ensure transactional integrity, you might consider using interactive transactions to perform this operation.


If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
